apps/web/src/components/Shared/Login/WalletSelector.tsx

```diff
@@ -118,6 +126,14 @@ const WalletSelector: FC<WalletSelectorProps> = ({
         message: challenge?.data?.challenge?.text
       });
 
+      if (status === 'success' && !!signer) {
+        // Get XMTP keys & Store it in index
+        const keys = await Client.getKeys(signer, { env: XMTP_ENV });
+        setKeys(keys);
+
+        await initialize({ keys, options: { env: XMTP_ENV }, signer });
+      }
+
```

Comment:
never ask for signing a login, all users not gonna use DMs, move it on messages page load!